src/components/Insights.tsx

```diff
@@ -34,41 +35,74 @@ export default function Insights({ isAuthorized, data, updateEntryDate, selectio
     }
   }, [isAuthorized, router]);
 
-  // Fetch page data! (relies on `date`, only available on the client)
   useEffect(() => {
-    // Only run the side-effect ONCE per chosenDate
     if (loadOnce.current === chosenDate) {
       return;
     }
+
     loadOnce.current = chosenDate;
 
-    async function loadPageData() {
-      setLoading(true);
-      try {
-        const response = await getInsights(chosenDate);
-        if (response.insights) {
-          setInsights(response.insights);
-        }
-        else if (response.horoscope) {
-          setHoroscope(response.horoscope);
-        }
-        else {
-          setEmptyDay(true);
-        }
-        setLoading(false);
-      } catch (err) {
-        setError(true);
-        console.error(err);
+    loadPageData();
+    getFormData();
+
+    return () => { 
+      setError(false);
+   }
+
+  }, [user, date, chosenDate, setSelections, router, data]);
+
+  async function loadPageData() {
+    setLoading(true);
+    try {
+      const response = await getInsights(chosenDate);
+      if (response.insights) {
+        setInsights(response.insights);
+      }
+      else if (response.horoscope) {
+        setHoroscope(response.horoscope);
       }
+      else {
+        setEmptyDay(true);
+      }
+      setLoading(false);
+    } catch (err) {
+      setError(true);
+      console.error(err);
     }
-    loadPageData();
-  }, [user, date, chosenDate]);
+  }
+
+  const getFormData = async () => {
+    setLoading(true);
+    try {
+      const entryInfo = await getEntry(false, formatDateForDB(convertStringToDate(chosenDate)), data ? data[0].passage_user_id : '');
+
+      if (entryInfo.data) {
+        setSelections({ FLOW: entryInfo.data.flow, MOOD: entryInfo.data.mood, CRAVINGS: entryInfo.data.craving });
+      } else {
+        setSelections({ FLOW: null, MOOD: null, CRAVINGS: null });
+      }
+    } catch (err) {
+      if (err instanceof Error) setError(true);
+    }
+    setLoading(false);
+  };
```

Comment:
It looks like this should probably be defined in the `useEffect` too. (we want to be sure to add all of the `useEffect`'s dependencies inside the dependency array

```./src/components/Insights.tsx
52:6  Warning: React Hook useEffect has missing dependencies: 'getFormData' and 'loadPageData'. Either include them or remove the dependency array.  react-hooks/exhaustive-deps
```